common/djangoapps/custom_reports/utils.py

```diff
@@ -0,0 +1,286 @@
+from threading import Thread
+from grades.course_data import CourseData
+from grades.course_grade import CourseGrade
+from custom_reports.api import ENABLE_THREADING_IN_RG, TOTAL_THREADS_IN_WEEKLY_REPORT
+from opaque_keys.edx.keys import CourseKey
+from student.models import CourseEnrollment
+from datetime import timedelta, date
+from courseware import courses
+from django.contrib.auth.models import User
+from .models import ReportsStatus
+import requests
+import uuid, csv, json, math, copy
+from api import REPORTS_DIR
+from django.conf import settings
+from django.core.validators import URLValidator
+import logging
+
+AUDIT_LOG = logging.getLogger("audit")
+
+class CompletionCheckThread(Thread):

```

Comment:
Why don't use tasks with celery ??

Reply:
I'm guessing they don't know about celery. I don't.

Is this an acceptable approach anyway?